doc/nrf/libraries/others/uart_async_adapter.rst

```diff
@@ -0,0 +1,103 @@
+The sample of such a driver is USB CDC ACM driver.
+It provides Pooling API and Interrupt-driven API, not implementing Asynchronous API.
+The UART Async Adapter is initialized with such an UART device and communicates with it using Interrupt-driven API, providing Asynchronous API for the application.
+
+The reason for is is to remove the need to implement two ways of interfacing UART devices in software.
+By the usage of UART Async Adapter software needs only to implement interraction with UART device by the Asynchronous API and just initialize the adapter for the ones that does not support it.
+
+Requirements
+************
+
+Enabling UART Async Adapter would also enable :kconfig:option:`CONFIG_SERIAL_SUPPORT_ASYNC` and this is a hard requirement becouse without UART Async API support enabled the UART Async Adapter would not even compile.
+It is also required to enable :kconfig:option:`CONFIG_SERIAL_SUPPORT_INTERRUPT` option.
+
+Configuration
+*************
+
+Use the :kconfig:option:`CONFIG_UART_ASYNC_ADAPTER` Kconfig option to enable the library in the build system.
+
+Usage
+*****
+
+To use the UART Async Adapter use the following code pattern:
+
+.. code-block:: c
+
+    #include <uart_async_adapter.h>
+
+    ...
+    /* Get the UART device we want to use */
+    static const struct device *uart = DEVICE_DT_GET([UART device name]);
+
+    /* Create UART Async Adapter instance */
+    UART_ASYNC_ADAPTER_INST_DEFINE(async_adapter);
+
+    /* UART initialization (called before any UART usage) */
+    static int uart_init(void)
+    {
+        if (!uart_test_async_api(uart)) {
+		/* Implement API adapter */
+		uart_async_adapter_init(async_adapter, uart);
+		uart = async_adapter;
+	}
+	/* Continue initialization using Asynchronous API on uart device */
+	(...)
+    }
+
+Note that using the adapter in a way shown above the adapter usage it totally transparent for the application.
+If the selected UART device provides Asynchronous API, not adapter is initialized.
+If the selected UART device does not provide Asynchronous API, the adapter is initialized to use such a device and then the direct pointer to the device is replaced by the adapter.
+
+Samples using the library
+*************************
+
+The following |NCS| samples use this library:
+
+* :ref:`peripheral_uart`
+
+Limitations
+***********
+
+The usage of the adapter adds code and performance overhead over direct usage of UART Asynchronous API provided by the driver.
+For the UART drivers that are only providing Interrupt-driven API it is also wise to consider using it directly in the application.
+The real usage of this adapter would be to support multiple UART devices with only support for one API inside application, speeding up the development process in a cost of performance and code size.
+
+Dependencies
+************
+
+This module may use Zephyr logging subsystem.
```

Comment:
Link to the subsystem? Maybe :ref:`Zephyr logging API <zephyr:logging_api>`? Or something else?